src/app/public/modules/toggle/toggle.component.html

```diff
@@ -0,0 +1,21 @@
+<label
+  class="sky-toggle-wrapper sky-switch"
+  [ngClass]="{ 'sky-switch-disabled': disabled }">
```

Comment:
@Blackbaud-MattGregg Unless, having the checkbox communicates the "toggle" state by itself?

Reply:
@Blackbaud-SteveBrush Haven't seen a toggle role before, but yet the switch role could be a good solution for getting the semantics right for the case where there is a label and the states are on/off. I think there are a few other items to work out accessibility-wise. Before that, it's probably good to confirm with @Blackbaud-AdamFunderburk on the different means of use this toggle is intended to support in the UI. He and I discussed this a bit in terms of the label. 

The documentation I've seen indicates there are two different semantic structures that can be used. 

1. The label is visible next to the toggle and the state of the toggle is on/off. This is the one which I think would be the straightforward switch role on a button
2. The text next to the toggle is the state and the label would either be referenced off-screen/in aria-label attribute or a pointer to the label somewhere else on the screen with aria-labelledby.

I do think going down the route of using a button as opposed to a checkbox makes sense in our system b/c these are inputs used to make a real-time change not in a modal form. Hence, this can be communicated differently than a checkbox doesn't activate/change system state anyplace else. 

This is a good article on the subject which I think could be useful here https://inclusive-components.design/toggle-button/

One other issue I noticed when I ran it was focus state not being indicated visually.

Reply:
Here's the current draft of the documentation for "Toggle". https://blackbaud.invisionapp.com/share/SEQZKL5YVPG#/368458068_toggle_Switch_Design

Most relevant is this section: "If the toggle represents exactly the states On and Off, the text label is not required. Use text labels for all other states. For example, if the options are more accurately described as Active and Inactive, use a text label."

Full disclosure, I've copied this nuance directly from the original specs for this component, which now have a couple years of dust on them. I don't personally know if the semantic hair-splitting in the guidelines here is affording any value in "simplicity" or not. I wouldn't put up a strong objection to a proposal to change the guidance to using text labels in all cases, if that would make things easier for accessibility (or anything else, really).

Re: focus styling, I'll experiment a bit, but our `sky-field-status(active)` treatment is probably a fine choice.

Re: cursor, what's our position on using `cursor: not-allowed` for the disabled state (and the disabled states of other controls, for that matter)?

Reply:
That verbiage is using "text label" which is what I think is confusing things. That "text label" (e.g., on/off, open/closed, etc.) is the state represented as text which someone visually can understand as the state along with the visual indication. What I'm talking about is the programatic label (i.e., what is open or closed, or what is on/off).
In the documentation, the example with "Automatically adjust..." is the label in that case and the button has a programmatic state for on/off or pressed/unpressed. In the example under it with the text next to the button (closed or open), closed/open isn't the label. It's the state. Having people hear Closed unpressed or Open pressed is going to be confusing.